### ISSUE TITLE
Use an interface on bot to allow mocking and unit testing

### ISSUE THREAD

**Issue by christianrondeau** [BeepBoopHQ/go-slackbot#5]
TL;DR: I'd like a BotInterface :)

I'm trying out go-slackbot, and as a TDD user, I want to test my handlers. Right now, I have two choices; either I wrap the https://github.com/BeepBoopHQ/go-slackbot/blob/master/bot.go type in my own, or I create a `SomethingHandler` than then calls a `SomethingHandlerImpl`, which abstracts away the `slackbot.Bot` type.

I think a simpler solution would be to reference to an `interface` instead of a concrete `struct` type, and add a `New` overload that receives the bot. Something like:

```
func New(slackToken string) *BotInterface {
	b := &Bot{Client: slack.New(slackToken)}
	return b
}

interface BotInterface interface {
    // This is what would be passed around
}

type Bot struct {
    // Nothing else changes...
}
```

I could just make the interface myself, but then I also need to mock `MessageHandler` (because https://github.com/BeepBoopHQ/go-slackbot/blob/58027c0d5238a0604305652ea526b3c18a7cead1/types.go#L19 references directly the concrete type), so it becomes complicated :)

The code might not be perfect, but you get the idea. I can then do:

```
func TestHelloHandler(t *testing.T) {
    ctx := context.TODO()
    bot := MyBotMock{}
    evt := slack.MessageEvent{}

    HelloHandler(ctx, &bot, &evt)

    // Some assertion...
}
```

What do you think?
